src/realm/table.cpp

```diff
@@ -3257,8 +3257,6 @@ void Table::set_binary(size_t col_ndx, size_t ndx, BinaryData value, bool is_def
         throw LogicError(LogicError::column_index_out_of_range);
     if (!is_nullable(col_ndx) && value.is_null())
         throw LogicError(LogicError::column_not_nullable);
-    if (REALM_UNLIKELY(value.size() > ArrayBlob::max_binary_size))
-        throw LogicError(LogicError::binary_too_big);
```

Comment:
I have sync working with big changesets in my branch. 
We need to disable the limit both here and in the transaction log.
If we do not disdable it in the transaction log, the user cannot have a single
binary larger than 16 MB in sync, even though the sum of instructions in a changeset
could exceed 16MB. To get completely rid of the limit, we need to remove it in transact_log.hpp as well.

What is you plan for set_binary versus set_binary_big? 
Will there be two functions eventually?